### PR TITLE
Add parameter to use a different authentication digest hash

### LINKED ISSUE #3: Possible incompatibility issue with newer HPE SAN systems
This script works fine with HP MSA 2050 SA, but on a brand new HPE MSA 2060 SAN output is
`  [CRITICAL] login failed: Error: Authentication Unsuccessful`
I double checked the credentials and even the Admin credentials won't work even though login is possible to the HP Web GUIl

Could it be that HPE has changed API specifications on this new model so that this wonderful check script won't work any more?

### PR DESCRIPTION
**Pull request by RincewindsHat**
Fixes #3

### CODE REVIEW

**Comment by RincewindsHat** [comment]
@geotekberlin Could you try this?

**Comment by RincewindsHat** [comment]
Fixes #3

File: check_hp_msa.py [hunk 1]
```diff
@@ -385,6 +388,10 @@ def parse_args():
     args.add_argument('--insecure', help='Do not check certificates', action='store_true')
 
     args.add_argument('--version', '-V', help='Print version', action='store_true')
+    args.add_argument('--authentication_hash_algo',
+                      help='The Hash algorithm to use for the authentication procedure ',
```
Comment (martialblog) [inline check_hp_msa.py:392]: Maybe mention that md5 is the default

File: check_hp_msa.py [hunk 2]
```diff
@@ -88,19 +88,22 @@ def __init__(self, api, username, password, logger=None, insecure=False):
 
         self.logger = logger
 
-
-    def credential_hash(self):
+    def credential_hash(self, hash_type=None):
         """
         Build a MD5 hashed credential built from username_password
```
Comment (martialblog) [inline check_hp_msa.py]:
```suggestion
        Build a hashed credential built from username_password
```

File: check_hp_msa.py [hunk 3]
```diff
@@ -385,6 +388,10 @@ def parse_args():
     args.add_argument('--insecure', help='Do not check certificates', action='store_true')
 
     args.add_argument('--version', '-V', help='Print version', action='store_true')
+    args.add_argument('--authentication_hash_algo',
```
Comment (martialblog) [inline check_hp_msa.py]: Can we make this something shorter? `authentication-hash` or `auth-hash-type`

Also, dashes in the Flag are translated by the argpase to `_`

> Any internal - characters will be converted to _ characters

Comment (RincewindsHat) [reply check_hp_msa.py]: better?

**Review by martialblog — APPROVED** [review]
Nice work

**Comment by geotekberlin** [comment]
I can confirm that this works on our HPE MSA 2060. Thanks a lot! :-)

------ Originalnachricht ------
Von "Markus Opolka" ***@***.***>
An "NETWAYS/check_hp_msa" ***@***.***>
Cc "geotekberlin" ***@***.***>; "Mention" ***@***.***>
Datum 17.10.2023 12:49:55
Betreff Re: [NETWAYS/check_hp_msa] Add parameter to use a different 
authentication digest hash (PR #4)

**Pull request merged**